st7920.py

```diff
@@ -128,7 +130,17 @@ def put_text(self, s, x, y):
 				pass
 			x += cw
 	
-	def redraw(self, dx1=0, dy1=0, dx2=127, dy2=63):
-		for i in range(dy1, dy2+1):
-			self.send(0,0,[0x80 + i%32, 0x80 + ((dx1/16) + (8 if i>=32 else 0))]) # set address
-			self.send(1,0,self.fbuff[i][dx1/8:(dx2/8)+1])
+	def redraw(self, dx1=0, dy1=0, dx2=127, dy2=63, full=False):
+		if full or self.currentlydisplayedfbuff == None:
+			for i in range(dy1, dy2+1):
+				self.send(0,0,[0x80 + i%32, 0x80 + ((dx1//16) + (8 if i>=32 else 0))]) # set address
+				self.send(1,0,self.fbuff[i][dx1//8:(dx2//8)+1])

```

Comment:
I think it would be really nice if these two lines were broken out into a separate function, rather than repeated below. Maybe `_send_line(self, row, dx1, dx2)`.